internal/beacon/beacon.go

```diff
@@ -11,15 +11,17 @@ import "net"
 type recv struct {
 	data []byte
 	src  net.Addr
+	done chan struct{}
 }
 
 type Interface interface {
 	Send(data []byte)
-	Recv() ([]byte, net.Addr)
+	Recv() ([]byte, net.Addr, chan struct{})
```

Comment:
Does the beacon changes really buy us very much, given that we're looking at like one small buffer every thirty seconds or so?